2.Arrays/arrays.js

```diff
@@ -0,0 +1,102 @@
+function isArray1(arr) {
+	return arr instanceof Array;
+}
+
+console.log(isArray1([3,null,undefined,4]));
+
+function isArray2(arr) {
+	return Array.isArray(arr);
+}
+
+console.log(isArray2([3,null,undefined,4]));
+console.log(isArray2({0:2, 1:3}));
+
+function range(from, to, step) {
+	var arr = [];
+	if (to === undefined) {
+		to = from;
+		from = 0;
+	}
+	(step === undefined) && (step = 1);
+	for(let i = from; i < to; i += step) {
+		arr.push(i);
+	}
+	return(arr);
+}
+
+console.log(range(1,10,2));
+
+function compactCycle(arr) {
+	var newArr = [];
+	for (let i = 0; i < arr.length; i++) {
+		(arr[i] == true) && (newArr.push(arr[i]));
```

Comment:
1. brackets are not required here
2.  `arr[i] && newArr.push(arr[i])` should work